--- a/src/linguist/CMakeLists.txt
+++ b/src/linguist/CMakeLists.txt
@@ -22,7 +22,7 @@ add_subdirectory(lupdate-pro)
 add_subdirectory(ltext2id)
 if(QT_FEATURE_process AND QT_FEATURE_pushbutton AND QT_FEATURE_toolbutton
     AND QT_FEATURE_png AND QT_FEATURE_mdiarea AND QT_FEATURE_syntaxhighlighter
-    AND TARGET Qt::Widgets AND TARGET Qt::Quick)
+    AND TARGET Qt::Widgets)
     add_subdirectory(linguist)
 endif()
 
--- a/src/linguist/linguist/CMakeLists.txt
+++ b/src/linguist/linguist/CMakeLists.txt
@@ -93,14 +93,6 @@ qt_internal_extend_target(linguist CONDITION QT_FEATURE_printsupport
         Qt::PrintSupport
 )
 
-qt_internal_extend_target(linguist CONDITION QT_BUILDING_QT
-    SOURCES
-        qmlformpreviewview.h qmlformpreviewview.cpp
-    LIBRARIES
-        Qt::QuickWidgets
-        Qt::QuickLayouts
-)
-
 if(QT_BUILDING_QT)
     set(available_plugins "")
 
--- a/src/linguist/linguist/mainwindow.cpp
+++ b/src/linguist/linguist/mainwindow.cpp
@@ -14,9 +14,6 @@
 #include "finddialog.h"
 #include <helpclient.h>
 #include "uiformpreviewview.h"
-#ifndef Q_OS_WASM
-#  include "qmlformpreviewview.h"
-#endif // Q_OS_WASM
 #include "globals.h"
 #include "messageeditor.h"
 #include "messagemodel.h"
@@ -74,13 +71,6 @@ static bool hasUiFormPreview(const QString &fileName)
     return fileName.endsWith(".ui"_L1) || fileName.endsWith(".jui"_L1);
 }
 
-#ifndef Q_OS_WASM
-static bool hasQmlFormPreview(const QString &fileName, bool qmlPreviewChecked)
-{
-    return fileName.endsWith(QLatin1String(".qml")) && qmlPreviewChecked;
-}
-#endif // Q_OS_WASM
-
 static const int MessageMS = 2500;
 
 } // namespace
@@ -357,10 +347,6 @@ MainWindow::MainWindow(HelpClientType helpClientType)
     m_sourceAndFormDock->setWidget(m_sourceAndFormView);
     m_uiFormPreviewView = new UiFormPreviewView(0, m_dataModel);
     m_sourceCodeView = new SourceCodeView(0);
-#ifndef Q_OS_WASM
-    m_qmlFormPreviewView = new QmlFormPreviewView(m_dataModel);
-    m_sourceAndFormView->addWidget(m_qmlFormPreviewView);
-#endif // Q_OS_WASM
     m_sourceAndFormView->addWidget(m_sourceCodeView);
     m_sourceAndFormView->addWidget(m_uiFormPreviewView);
 
@@ -571,9 +557,6 @@ void MainWindow::modelCountChanged()
     m_ui.actionFindPrev->setEnabled(false);
 
     m_uiFormPreviewView->setSourceContext(-1, 0);
-#ifndef Q_OS_WASM
-    m_qmlFormPreviewView->setSourceContext(-1, 0);
-#endif // Q_OS_WASM
     updateVisibleColumns();
 }
 
@@ -1614,11 +1597,6 @@ void MainWindow::translationChanged(const MultiDataIndex &index)
     MessageItem *m = m_dataModel->messageItem(index);
     if (hasUiFormPreview(m->fileName()))
         m_uiFormPreviewView->setSourceContext(index.model(), m);
-#ifndef Q_OS_WASM
-    else if (hasQmlFormPreview(m->fileName(), m_ui.actionQmlPreview->isChecked()))
-        if (!m_qmlFormPreviewView->setSourceContext(index.model(), m))
-            m_ui.actionQmlPreview->setChecked(false);
-#endif // Q_OS_WASM
 }
 
 // This and the following function operate directly on the messageitem,
@@ -1634,12 +1612,6 @@ void MainWindow::updateTranslation(const QStringList &translations)
     m->setTranslations(translations);
     if (!m->fileName().isEmpty() && hasUiFormPreview(m->fileName()))
         m_uiFormPreviewView->setSourceContext(m_currentIndex.model(), m);
-#ifndef Q_OS_WASM
-    else if (!m->fileName().isEmpty()
-             && hasQmlFormPreview(m->fileName(), m_ui.actionQmlPreview->isChecked()))
-        if (!m_qmlFormPreviewView->setSourceContext(m_currentIndex.model(), m))
-            m_ui.actionQmlPreview->setChecked(false);
-#endif // Q_OS_WASM
     updateDanger(m_currentIndex, true);
 
     if (m->isFinished())
@@ -2068,11 +2040,7 @@ void MainWindow::setupMenuBar()
     connect(m_ui.actionDisplayGuesses, &QAction::triggered,
             m_phraseView, &PhraseView::toggleGuessing);
     connect(m_ui.actionStatistics, &QAction::triggered, this, &MainWindow::showStatistics);
-#ifndef Q_OS_WASM
-    connect(m_ui.actionQmlPreview, &QAction::triggered, this, &MainWindow::toggleQmlPreview);
-#else
     m_ui.actionQmlPreview->setVisible(false);
-#endif // Q_OS_WASM
     connect(m_ui.actionVisualizeWhitespace, &QAction::triggered,
             this, &MainWindow::toggleVisualizeWhitespace);
     connect(m_ui.actionIncreaseZoom, &QAction::triggered,
@@ -2182,15 +2150,7 @@ void MainWindow::updateSourceView(int model, MessageItem *item)
         if (hasUiFormPreview(item->fileName())) {
             m_sourceAndFormView->setCurrentWidget(m_uiFormPreviewView);
             m_uiFormPreviewView->setSourceContext(model, item);
-#ifndef Q_OS_WASM
-        } else if (hasQmlFormPreview(item->fileName(), m_ui.actionQmlPreview->isChecked())
-                   && m_qmlFormPreviewView->setSourceContext(model, item)) {
-            m_sourceAndFormView->setCurrentWidget(m_qmlFormPreviewView);
-#endif // Q_OS_WASM
         } else {
-#ifndef Q_OS_WASM
-            m_ui.actionQmlPreview->setChecked(false);
-#endif // Q_OS_WASM
             m_sourceAndFormView->setCurrentWidget(m_sourceCodeView);
             QDir dir = QFileInfo(m_dataModel->srcFileName(model)).dir();
             QString fileName = QDir::cleanPath(dir.absoluteFilePath(item->fileName()));
@@ -2795,16 +2755,6 @@ void MainWindow::showStatistics()
     updateStatistics();
 }
 
-#ifndef Q_OS_WASM
-void MainWindow::toggleQmlPreview()
-{
-    if (m_ui.actionQmlPreview->isChecked())
-        m_sourceAndFormView->setCurrentWidget(m_qmlFormPreviewView);
-    else
-        m_sourceAndFormView->setCurrentWidget(m_sourceCodeView);
-}
-#endif // Q_OS_WASM
-
 void MainWindow::toggleVisualizeWhitespace()
 {
     m_messageEditor->setVisualizeWhitespace(m_ui.actionVisualizeWhitespace->isChecked());
--- a/src/linguist/linguist/mainwindow.h
+++ b/src/linguist/linguist/mainwindow.h
@@ -37,9 +37,6 @@ class ErrorsView;
 class FocusWatcher;
 class HelpClient;
 class UiFormPreviewView;
-#ifndef Q_OS_WASM
-class QmlFormPreviewView;
-#endif // Q_OS_WASM
 class MessageEditor;
 class PhraseView;
 class SourceCodeView;
@@ -141,9 +138,6 @@ private slots:
                   FindDialog::FindOptions options, int statusFilter);
     void revalidate();
     void showStatistics();
-#ifndef Q_OS_WASM
-    void toggleQmlPreview();
-#endif // Q_OS_WASM
     void toggleVisualizeWhitespace();
     void onWhatsThis();
     void updatePhraseDicts();
@@ -219,9 +213,6 @@ private:
     QTabWidget *m_contextAndLabelView;
     SourceCodeView *m_sourceCodeView;
     UiFormPreviewView *m_uiFormPreviewView;
-#ifndef Q_OS_WASM
-    QmlFormPreviewView *m_qmlFormPreviewView;
-#endif // Q_OS_WASM
     ErrorsView *m_errorsView;
     QLabel *m_progressLabel;
     QLabel *m_modifiedLabel;
